src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -138,24 +138,34 @@ private static async Task<string> GetTestData(string testDataPath, ScriptJobHost
         private static Uri GetFunctionHref(string functionName, string baseUrl) =>
             new Uri($"{baseUrl}/admin/functions/{functionName}");
 
-        private static Uri GetFunctionInvokeUrlTemplate(string baseUrl, FunctionMetadata functionMetadata, string routePrefix)
+        internal static Uri GetFunctionInvokeUrlTemplate(string baseUrl, FunctionMetadata functionMetadata, string routePrefix)
         {
             var httpBinding = functionMetadata.InputBindings.FirstOrDefault(p => string.Compare(p.Type, "httpTrigger", StringComparison.OrdinalIgnoreCase) == 0);
-            string template = null;
+
             if (httpBinding != null)
             {
+                string customRoute = null;
                 if (httpBinding.Raw != null && httpBinding.Raw.TryGetValue("route", StringComparison.OrdinalIgnoreCase, out JToken value))
                 {
                     // a custom route is specified
-                    template = (string)value;
+                    customRoute = (string)value;
+                }
+
+                string uriString = baseUrl.TrimEnd('/');
+                if (!string.IsNullOrEmpty(routePrefix))
+                {
+                    uriString += $"/{routePrefix.TrimEnd('/')}";
+                }
+
+                if (!string.IsNullOrEmpty(customRoute))
```

Comment:
There was actually a bug in this code - if a custom route was being used, it was being appended directly to the base route, w/o including the route prefix.